core-bundle/src/Twig/Runtime/HighlightResult.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\Twig\Runtime;
+
+use Highlight\HighlightResult as BaseHighlightResult;
+
+/**
+ * This class is a thin wrapper around @see \Highlight\HighlightResult that
+ * provides an additional __toString() function.
+ */
+class HighlightResult extends BaseHighlightResult implements \Stringable
+{
+    /**
+     * @internal
+     */
+    public function __construct(BaseHighlightResult|\stdClass $result)
```

Comment:
In case you're wondering why we've got `stdClass` in the type declaration: the highlighter component creates a `stdClass` shaped like a `BaseHighlightResult`, but it isn't actually an instance of one (duck typing).